### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by miltontony** [comment]
@codiebeulaine tests now running on travis ;)

**Comment by miltontony** [comment]
@smn review please

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: python
+python:
+  - "2.7"
+install:
+  - "pip install coveralls coverage --use-wheel"
+  - "pip install -e . --use-wheel"
```
Comment (smn) [inline .travis.yml]: `--use-wheel` is now the default, just put `pip install --upgrade pip` as the first install to make sure you always have the latest.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+here = os.path.abspath(os.path.dirname(__file__))
+with open(os.path.join(here, 'README.rst')) as f:
+    README = f.read()
+
+with open(os.path.join(here, 'requirements.txt')) as f:
+    requires = filter(None, f.readlines())
+
+setup(
+    name='unicore-hub',
+    version='1.0.0',
```
Comment (smn) [inline setup.py]: I've found the `VERSION` stuff we're using in other repos to work quite well. Hooking this up with Travis' deploy to pypi stuff ++

File: sso/permissions.py [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
-    groups = user.groups.values_list('name', flat=True)
+    service_domain = urlparse(service).netloc
 
-    if AuthorizedSite.objects.filter(site=service).exists():
-        website = AuthorizedSite.objects.get(site=service)
-        site_groups = website.group.values_list('name', flat=True)
+    matched_sites = [
+        s for s in AuthorizedSite.objects.all()
+        if re.match(
+            re.compile(s.site.replace('.', '\\.').replace('*', '.*')),
```
Comment (smn) [inline sso/permissions.py]: https://docs.python.org/2/library/re.html#re.escape is probably useful here?

Comment (smn) [inline sso/permissions.py]: Also:

``` python
>>> from glob.fnmatch import fnmatch
>>> fnmatch.fnmatch('http://bar.foo.com', 'http://*.foo.com')
True
>>>
```

Manually escaping regex in any code that relates to auth makes me quite nervous.

File: unicoresso/tests/test_permissions.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        user.groups.add(group)
+        user.save()
+
+        site = AuthorizedSite.objects.create(
+            site='cms.za.mama.qa-hub.unicore.io')
+        site.group.add(group)
+        site.save()
+
+        attr = permissions.custom_attributes(
+            user, 'htto://cms.za.mama.qa-hub.unicore.io/login/')
```
Comment (smn) [inline unicoresso/tests/test_permissions.py]: How can `htto` pass?

**Comment by smn** [comment]
:+1:

**Pull request merged**